src/dialects/db2/data-types.ts

```diff
@@ -0,0 +1,380 @@
+        // long = 2GB
+        return 'BLOB(2G)';
+      }
+
+      return `BLOB(${this._length})`;
+    }
+
+    return 'BLOB'; // 1MB
+  }
+
+  static escape(blob: AcceptableTypeOf<BaseTypes.BLOB>, options: StringifyOptions) {
+    return `BLOB('${options.escape(blob.toString())}')`;
+  }
+
+  protected _stringify(value: AcceptableTypeOf<BaseTypes.BLOB>, options: StringifyOptions) {
+    if (Buffer.isBuffer(value)) {
+      return `BLOB('${options.escape(blob.toString())}')`;
+    }
+
+    if (Array.isArray(value)) {
+      value = Buffer.from(value);
+    } else {
+      value = Buffer.from(value.toString());
+    }
+
+    const hex = value.toString('hex');
+
+    return this._hexify(hex);
+  }
+
+  protected _hexify(hex: string) {
+    return `x'${hex}'`;
+  }
+}
+
+export class STRING extends BaseTypes.STRING {
+  static readonly escape: false;
+
+  toSql() {
+    if (!this._binary) {
+      if (this._length <= 4000) {
+        return `VARCHAR(${this._length})`;
+      }
+
+      return `CLOB(${this._length})`;
+    }
+
+    if (this._length < 255) {
+      return `CHAR(${this._length}) FOR BIT DATA`;
+    }
+
+    if (this._length <= 4000) {
+      return `VARCHAR(${this._length}) FOR BIT DATA`;
+    }
+
+    return `BLOB(${this._length})`;
+  }
+
+  protected _hexify(hex: string) {
+    return `x'${hex}'`;
+  }
+
+  protected _stringify(
+    value: AcceptableTypeOf<BaseTypes.STRING>,
+    options: StringifyOptions,
+  ) {
+    if (this._binary) {
+      return this._hexify(value.toString('hex'));
+    }
+
+    return options.escape(value.toString());
+  }
+
+  protected _bindParam(
+    value: AcceptableTypeOf<BaseTypes.STRING>,
+    options: BindParamOptions,
+  ) {
+    return options.bindParam(this._binary ? Buffer.from(value) : value);
+  }
+}
+
+export class TEXT extends BaseTypes.TEXT {
+  // @ts-expect-error missing call to super() (useless here)
+  constructor() {
+    throw new Error('DB2 does not support the TEXT data-type');
+  }
+}
+
+export class BOOLEAN extends BaseTypes.BOOLEAN {
+  toSql() {
+    return 'BOOLEAN';
+  }
+
+  protected _sanitize(value: RawTypeOf<BaseTypes.BOOLEAN>) {
+    if (value !== null && value !== undefined) {
+      if (Buffer.isBuffer(value) && value.length === 1) {
+        // Bit fields are returned as buffers
+        value = value[0];
+      }
+
+      if (typeof value === 'string') {
+        // Only take action on valid boolean strings.
+        value = value === 'true' ? true : value === 'false' ? false : value;
+        value = value === '\u0001' ? true : value === '\u0000' ? false : value;
+      } else if (typeof value === 'number') {
+        // Only take action on valid boolean integers.
+        value = value === 1 ? true : value === 0 ? false : value;
+      }
+    }
+
+    return typeof value === 'boolean' ? value : null;
+  }
+
+  static readonly parse = this.prototype._sanitize;
+}
+
+export class UUID extends BaseTypes.UUID {
+  toSql() {
+    return 'CHAR(36) FOR BIT DATA';
```

Comment:
Not strictly DB2 related: This seems to be what we do in other dialects without proper UUID support too, but ideally we'd store it as a 16 byte buffer / binary instead.

This would reduce the size of uuids by >50% and in theory speed things up a lil' too.

Reply:
Sounds good to me :)